### PR TITLE
Reduce dependencies for updates, switch to Ubuntu 22.04

### PR DESCRIPTION
**Pull request by galewis2**
@Overv @Istador 

The most major things in this:

1. Avoid the extra call out to yet another service that may go down - why not just use the widely adopted approach for getting the sequence number to start updates from?
2. The timestamp header detected from the PBF may not exist in non-Geofabrik PBFs. Should rely on a more standard timestamp header.. and use better methods for accessing it
3. Enable using hourly/daily as well as different replication URLs
4. Enable using different max_intervals

Tested via:

```
sudo docker run \
    -e UPDATES=enabled \
    -e DOWNLOAD_PBF=https://download.geofabrik.de/europe/luxembourg-210101.osm.pbf \
    -e DOWNLOAD_POLY=https://download.geofabrik.de/europe/luxembourg.poly \
    -v openstreetmap-data:/var/lib/postgresql/12/main \
    -v openstreetmap-rendered-tiles:/var/lib/mod_tile \
    tiles \
    import

sudo docker run \
    -p 8080:80 \
    -e UPDATES=enabled \
    -v openstreetmap-data:/var/lib/postgresql/12/main \
    -v openstreetmap-rendered-tiles:/var/lib/mod_tile \
    -d tiles \
    run
```

### CODE REVIEW

**Comment by galewis2** [comment]
@Overv  Any thoughts on this?

**Comment by Overv** [comment]
@galewis2 I'll look at this soon.

**Comment by Overv** [comment]
Someone who is familiar with the automatic updating will have to review this. (I don't use that feature myself and I'm not very familiar with the mechanisms.)

**Comment by galewis2** [comment]
@Overv , @stevo01  Thanks!

**Comment by stevo01** [comment]
Hi all,
I replaced the update procedures of my tile server installation by a script (**osm2pgsql-replication** ) that comes with the osm2psql version >=1.5 in the meantime. 

see https://osm2pgsql.org/doc/manual.html#updating-an-existing-database for more details.

**Comment by galewis2** [comment]
Hrm, I will look at switching to the osm2pgsql-replication script as it looks like that will cut down some of the code in this repo.

**Comment by galewis2** [comment]
@Overv  @stevo01 
The more I looked at it, the way updates are already handled in this repo is how it's done in Switch2OSM and seems to be the most common and accepted way. I don't think it makes sense to change from the usual approach.

I'd even consider reverting back the update script in this repo back to the upstream one in https://github.com/SomeoneElseOSM/mod_tile.git to minimize the code to be maintained.

**Comment by galewis2** [comment]
@Overv  @stevo01 
The more I looked at it, the way updates are already handled in this repo is how it's done in Switch2OSM and seems to be the most common and accepted way. I don't think it makes sense to change from the usual approach.

I'd even consider reverting back the update script in this repo back to the upstream one in https://github.com/SomeoneElseOSM/mod_tile.git to minimize the code to be maintained.

File: openstreetmap-tiles-update-expire [hunk 1]
```diff
@@ -107,11 +110,18 @@ freelock()
...
-    mv $WORKOSM_DIR/configuration.txt $WORKOSM_DIR/configuration_orig.txt
-    sed "s!baseUrl=http://planet.openstreetmap.org/replication/minute!baseUrl=https://planet.openstreetmap.org/replication/minute!" $WORKOSM_DIR/configuration_orig.txt > $WORKOSM_DIR/configuration.txt
+    mkdir -p $WORKOSM_DIR
+    $OSMOSIS_BIN -v 5 --read-replication-interval-init workingDirectory=$WORKOSM_DIR 1>&2 2> "$OSMOSISLOG"
+
+    init_seq=$(pyosmium-get-changes --server $REPLICATION_URL -D $1)
+    url_dynamicPart=$(printf %09d $init_seq | sed 's_\([0-9][0-9][0-9]\)\([0-9][0-9][0-9]\)\([0-9][0-9][0-9]\)_\1/\2/\3_')
+    wget $REPLICATION_URL/$url_dynamicPart.state.txt -O $WORKOSM_DIR/state.txt
+
+        cat > /var/lib/mod_tile/.osmosis/configuration.txt <<- EOM
```
Comment (Istador) [inline openstreetmap-tiles-update-expire:120]:
```diff
@@ -120,1 +120,1 @@ if [ $# -eq 1 ] ; then
-        cat > /var/lib/mod_tile/.osmosis/configuration.txt <<- EOM
+    cat > $BASE_DIR/.osmosis/configuration.txt <<- EOM
```

**Comment by galewis2** [comment]
@Istador Sorry for the delay getting back at this.

I realized now after doing the conflict resolve that a lot of stuff was added to this PR in my absence, and it's diff is gnarly. At this point, I think it's best for me to close this PR, and re-open a new one, freshly branched off Master with just the changes necessary.

**Comment by Istador** [comment]
In case you're missing it (from the [test job log](https://github.com/Overv/openstreetmap-tile-server/runs/6921621389)):
```
+ '[' enabled == enabled ']'
++ cut -b35-44
++ grep osmosis_replication_timestamp=
++ osmium fileinfo /data/region.osm.pbf
+ REPLICATION_TIMESTAMP=2022-06-15
+ sudo -u renderer openstreetmap-tiles-update-expire.sh 2022-06-15
usage: pyosmium-get-changes [-h] [-v] [-o OUTFILE] [--format FORMAT]
                            [--server SERVER_URL] [--cookie COOKIE]
                            [-s OUTSIZE] [-I ID | -D DATE | -O OSMFILE]
                            [-f SEQ_FILE] [--ignore-osmosis-headers] [-d]
                            [--socket-timeout SOCKET_TIMEOUT] [--version]
pyosmium-get-changes: error: argument -D/--start-date: Date needs to be in ISO8601 format (e.g. 2015-12-24T08:08:08Z).
Error: Process completed with exit code 2.
```

**Comment by galewis2** [comment]
> In case you're missing it (from the [test job log](https://github.com/Overv/openstreetmap-tile-server/runs/6921621389)):
> 
> ```
> + '[' enabled == enabled ']'
> ++ cut -b35-44
> ++ grep osmosis_replication_timestamp=
> ++ osmium fileinfo /data/region.osm.pbf
> + REPLICATION_TIMESTAMP=2022-06-15
> + sudo -u renderer openstreetmap-tiles-update-expire.sh 2022-06-15
> usage: pyosmium-get-changes [-h] [-v] [-o OUTFILE] [--format FORMAT]
>                             [--server SERVER_URL] [--cookie COOKIE]
>                             [-s OUTSIZE] [-I ID | -D DATE | -O OSMFILE]
>                             [-f SEQ_FILE] [--ignore-osmosis-headers] [-d]
>                             [--socket-timeout SOCKET_TIMEOUT] [--version]
> pyosmium-get-changes: error: argument -D/--start-date: Date needs to be in ISO8601 format (e.g. 2015-12-24T08:08:08Z).
> Error: Process completed with exit code 2.
> ```

Yep literally just fixed it :)

**Comment by galewis2** [comment]
@Istador @Overv  @dbahrdt @nielsole So I went a little overboard with this after hitting a build issue.

So here's all the additional changes in here:

1. I switched to Ubuntu 22.04, so I could get mod_tile/renderd and others without compile. That chopped out a lot of code but required changing some paths to defaults
2. The default layer is called "default", not "ajt". That touched the readme as well
3. In order to use the default mod_tile and not forked version, I had to switch away from osmosis-db_replag, so I use dateutils.ddiff to calculate the lag in seconds
4. I enabled the use of skipping the requirement for a *.poly file
5. All the /var/log/tiles logs are now spit-out to stdout, so it's a lot easier to see what it's doing while it's updating

**Review by nielsole — APPROVED** [review]

**Comment by Istador** [comment]
Please add a new tag `v2.1.0` or so to the latest commit in `master`.
(Changing the underlying base image to a new major version is a big change, even if it might not break anything for users. If it does break compability with existing installations/volumes: release as `v3.0.0` instead.)

(BTW: the release tag [v2.0.1](https://github.com/Overv/openstreetmap-tile-server/releases/tag/v2.0.1) wrongly links to a comparison between `v2.0.0` and `v2.1.0` which doesn't exist yet.)

**Pull request merged**